lightning/impl/sgd_fast.pyx

```diff
@@ -53,6 +53,21 @@ cdef class ModifiedHuber(LossFunction):
         else:
             return 4.0 * y
 
+cdef class WeightedLoss(LossFunction):
+
+    cdef double[:] sample_weights
+    cdef LossFunction loss_func
+
+    def __init__(self, np.ndarray[double, ndim=1] sample_weights, LossFunction loss_func):
+        self.sample_weights = sample_weights
+        self.loss_func = loss_func
+
+    cpdef double loss(self, double p, double y, int i):
```

Comment:
I would say it depends on the performance hit. If it's something like 30%, your design is worth it. If it's more like 5%, I think I would prefer the simpler design.